File: ios/Plugin/PDFViewerPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(PDFViewerPlugin)
public class PDFViewerPlugin: CAPPlugin {
    private let implementation = PDFViewer()

    @objc func open(_ call: CAPPluginCall) {
        if let url = call.getString("url") {
            if let pdfURL = URL(string: url) {
                let top = call.getInt("top") ?? 0;
                
                implementation.open(pdfURL, top: top);
            }
        }
                
        call.resolve()
    }
    
    @objc func close(_ call: CAPPluginCall) {
        implementation.closeViewer();
        
        call.resolve()
    }
}

File: android/src/main/java/ch/nadlo/oss/capacitor/pdf_viewer/PDFViewerPlugin.java
package ch.nadlo.oss.capacitor.pdf_viewer;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "PDFViewer")
public class PDFViewerPlugin extends Plugin {

    private final PDFViewer implementation = new PDFViewer();

    @Override
    public void load() {
        super.load();

        implementation.setBridge(this.getBridge());
    }

    @PluginMethod
    public void open(PluginCall call) {
        String url = call.getString("url");
        String title = call.getString("title", "");

        if (url != null && !url.isEmpty()) {
            implementation.openViewer(url, title);
        }

        call.resolve();
    }

    @PluginMethod
    public void close(PluginCall call) {
        // implementation.close();

        call.resolve();
    }
}
